addons/sourcemod/scripting/AFKManager.sp

```diff
@@ -379,7 +394,7 @@ public Action Timer_CheckPlayer(Handle Timer, any Data)
 				ChangeClientTeam(client, CS_TEAM_SPECTATOR);
 			}
 		}
-		else if(g_fKickTime > 0.0 && (!g_iImmunity || g_iImmunity == 3 || !CheckAdminImmunity(client)))
+		else if(g_fKickTime > 0.0 && (!g_iImmunity || g_iImmunity == 3 || !CheckAdminImmunity(client)) || g_Players_bNotJoinedTeam[client])
```

Comment:
i dont think thats a good way to check if the player is in CS_TEAM_NONE

cant we just move the player in spectator if he is in CS_TEAM_NONE ?

Reply:
Sure. Nice idea. It will also make the code cleaner.
Will do it ASAP.